Fix `LngLatBounds.extend()` to Handle Object Coordinates

Buffalom opened the pull request:
This PR fixes `LngLatBounds.extend()` to handle `{ lng: number, lat: number }` and `{ lon: number, lat: number }` coordinates.

Fixes #2424.

## Launch Checklist

<!-- Thanks for the PR! Feel free to add or remove items from the checklist. -->


 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

HarelM commented on src/geo/lng_lat_bounds.ts:
```diff
@@ -103,7 +103,11 @@ class LngLatBounds {
                     return this.extend(LngLat.convert(lngLatObj));
                 }
+
+            } else if (typeof obj === 'object' && obj !== null) {
```
I think a better check would be to check that lng and lat exists in the object.

HarelM commented:
Thanks for taking the time to open this PR!
Please add a changelog entry as well, other than that and my eval comment I think it's good to go.

Buffalom commented:
Thanks for your quick review. I updated the code according to your comment and added the changelog entry. Let me know if there is anything else you would like me to do.
